### ISSUE TITLE
Weird spurious failures related to enabled APIs or max-surge

### ISSUE THREAD

**Issue by kosta** [google-github-actions/deploy-cloudrun#521]
### TL;DR

Weird spurious failures where max-surge is injected into the annotations or error API that need to be enabled (but are enabled)

### Expected behavior

Deploy should just work

### Observed behavior

Deployment fails with one of 2 seemingly unrelated error messages:
1. Skipped validating Cloud SQL API and Cloud SQL Admin API enablement due to an issue contacting the Service Usage  API. Please ensure the Cloud SQL API and Cloud SQL Admin API are activated
2. ERROR: (gcloud.run.services.update) metadata.annotations: Project is not allowed to use run.googleapis.com/max-surge annotation

About 1.: both APIs are enabled.
About 2.: _After_ this error, there is a `run.googleapis.com/max-surge: 0` annotation in the service yaml metadata that I then removed again.

### Action YAML

```yaml
      - name: "deploy service"
        id: deploy
        uses: "google-github-actions/deploy-cloudrun@v2"
        with:
          service: ${{inputs.app-name}}
          image: "${{vars.ARTIFACT_PATH}}/${{inputs.image-name}}"
          region: "europe-west3"
          flags: "--memory 4Gi --cpu 2 --execution-environment gen2 --no-cpu-throttling --allow-unauthenticated --service-account=${{vars.SERVICE_ACCOUNT_AUTOMAGICALLY}} --set-cloudsql-instances=${{vars.CLOUD_SQL_INSTANCE_NAME}} --max-instances=${{inputs.max-instances}} --min-instances=${{inputs.min-instances}}"
          env_vars: |
            SOME_ENV_VARS=removed
          secrets: |
            SOME_SECRETS=REMOVED:latest

            /secrets/on/filesytem=REMOVED_AS_WELL:latest

      - name: "update env variable on service"
        shell: bash
        run: "gcloud run services update ${{ inputs.app-name }} --region=europe-west3 --update-env-vars BASE_URL=${{ steps.base_url.outputs.BASE_URL }}"

      - name: "set revision traffic"
        id: revision_traffic
        uses: "google-github-actions/deploy-cloudrun@v2"
        with:
          service: ${{inputs.app-name}}
          region: "europe-west3"
          revision_traffic: "LATEST=100"
```

### Log output

```text
First run failed as "set revision traffic":

Running: gcloud run services update-traffic automagically --to-revisions LATEST=100 --format json --region europe-west3
Error: google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run services update-traffic automagically --to-revisions LATEST=100 --format json --region europe-west3`: Updating traffic...
failed
Updating traffic failed
ERROR: (gcloud.run.services.update-traffic) metadata.annotations: Project is not allowed to use run.googleapis.com/max-surge annotation

Note: After this run, the service actually had a run.googleapis.com/max-surge annotation which I removed.

Second run failed at "deploy service":

Running: gcloud run deploy automagically --image ... --update-env-vars ^d5^... --update-secrets ^d^... --update-labels ^43^... --format json --region europe-west3 --memory 4Gi --cpu 2 --execution-environment gen2 --no-cpu-throttling --allow-unauthenticated --service-account ... --set-cloudsql-instances ... --max-instances 100 --min-instances 1
Error: google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run deploy automagically --image ... --update-env-vars ^d5^... --update-secrets ^d^... --update-labels ^43^... --format json --region europe-west3 --memory 4Gi --cpu 2 --execution-environment gen2 --no-cpu-throttling --allow-unauthenticated --service-account ... --set-cloudsql-instances ... --max-instances 100 --min-instances 1`: Skipped validating Cloud SQL API and Cloud SQL Admin API enablement due to an issue contacting the Service Usage  API. Please ensure the Cloud SQL API and Cloud SQL Admin API are activated (see https://console.cloud.google.com/apis/dashboard).

Note: the APIs are both enabled

Third run failed at "update env variable on service":

Run gcloud run services update automagically --region=europe-west3 --update-env-vars BASE_URL=https://app.octomind.dev
Deploying...
failed
Deployment failed
ERROR: (gcloud.run.services.update) metadata.annotations: Project is not allowed to use run.googleapis.com/max-surge annotation
Error: Process completed with exit code 1.

Fourth attempt failed at "deploy service" again (but different error):

Run google-github-actions/deploy-cloudrun@v2
Successfully authenticated
Running: gcloud run deploy automagically --image ... --update-env-vars ^0d^... --update-secrets ^a0^... --update-labels ^b4^... --format json --region europe-west3 --memory 4Gi --cpu 2 --execution-environment gen2 --no-cpu-throttling --allow-unauthenticated --service-account ... --set-cloudsql-instances ... --max-instances 100 --min-instances 1
Error: google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run deploy automagically --image ... --update-env-vars ^0d^... --update-secrets ^a0^... --update-labels ^b4^... --format json --region europe-west3 --memory 4Gi --cpu 2 --execution-environment gen2 --no-cpu-throttling --allow-unauthenticated --service-account ... --set-cloudsql-instances ... --max-instances 100 --min-instances 1`: Skipped validating Cloud SQL API and Cloud SQL Admin API enablement due to an issue contacting the Service Usage  API. Please ensure the Cloud SQL API and Cloud SQL Admin API are activated (see https://console.cloud.google.com/apis/dashboard).
Deploying container to Cloud Run service [automagically] in project [...] region [europe-west3]
Deploying...
failed
Deployment failed
ERROR: (gcloud.run.deploy) metadata.annotations: Project is not allowed to use run.googleapis.com/max-surge annotation

Fifth run worked
```

### Additional information

We us this worflow fragment to deploy to three environments. For 2 of them, it works without issues. For the third one (prod, of course) we had the errors above, twice today.

**Comment by sethvargo** [comment]
Hi @kosta - thank you for opening an issue. Are you getting the same error if you run the outputted gcloud command? This looks like a runtime configuration error on your GCP project:

> metadata.annotations: Project is not allowed to use run.googleapis.com/max-surge annotation

Please note that this GitHub tracker is only for the GitHub Action. For general Cloud Run support, please see: https://cloud.google.com/run/docs/support

**Comment by kosta** [comment]
I do _not_ think it's a configuration error - or else the errors would not be so spurious and inconsistent.

However I think you're right. This plugin is only about invoking `gcloud` and it is `gcloud` that is behaving weirdly. So I will close this bug and either open a new bug with `gcloud` or talk to google support.
